src/init.cpp

```diff
@@ -418,4 +420,5 @@ void DynamicDataLoader::check_consistency()
     mission_type::check_consistency();
     item_action_generator::generator().check_consistency();
     harvest_list::check_consistency();
+    profession::check_consistency_json_item_substitution();
```

Comment:
That's even better. I think you can also get rid of `"json_item_substitution"` and simply check if `"substitutions"` is present.